prometheus-ksonnet/grafana/dashboards.libsonnet

```diff
@@ -7,11 +7,35 @@
     dashboard_config_maps: 8,
   },
 
-  // Extension point for you to add your own dashboards.
-  dashboards+:: {},
-  grafana_dashboards+:: {},
-  grafanaDashboards+:: $.dashboards + $.grafana_dashboards,
-  dashboardsByFolder+:: {},
+  // New API: Mixins go in the mixins map.
+  mixins+:: {},
+
+  // Legacy extension points for you to add your own dashboards.
+  local emptyMixin = {
+    grafanaDashboards+: {},
+  },
```

Comment:
What do we need this for? If you actually added a mixin here, it would be recursively merged with all other mixins, wouldn't it?

Reply:
Hence why its a local; prevents people adding any mixins here.

It is here so we can reliably do `mixin.grafanaDashboards` without having to check the field exists first.  Some mixins don't declare all the fields, and thats fine.

Reply:
Ah fine, the comment confused me